x-pack/plugins/task_manager/server/task_scheduling.ts

```diff
@@ -390,8 +390,8 @@ export class TaskScheduling {
             (taskInstance: OkResultOf<TaskLifecycleEvent>) => {
               // resolve if the task has run sucessfully
               if (isTaskRunEvent(taskEvent)) {
-                subscription.unsubscribe();
                 resolve(pick((taskInstance as RanTask).task, ['id', 'state']));
+                subscription.unsubscribe();
```

Comment:
Calling `subscription.unsubscribe()` before resolving the promise caused the `resolve` never to be called.